src/hsl_mc21.jl

```diff
@@ -10,10 +10,28 @@ export mc21
 # end
 
 function mc21ad(n, icn, licn, ip, lenr, iperm, numnz, iw)
-  ccall((:mc21ad_, libhsl_mc21),
-        Nothing,
-        (Ref{Int32}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}, Ptr{Int32}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}),
-         n         , icn       , licn      , ip        , lenr      , iperm     , numnz     , iw)
```

Comment:
@amontoison Why is `lenr` a `Ptr` and not a `Ref` like the other integers in this call?

Also, should we add some markup to prevent this kind of silly formatting?

Reply:
Yes, we should add some 
```
#! format: off
#! format: on
```
around the `ccall(s)`.

Reply:
I use `Ptr` because `lenr` is a vector.
![Capture d’écran du 2022-10-05 23-24-35](https://user-images.githubusercontent.com/35051714/194207336-31b99e76-24e0-40c6-bbfa-ec8248222e7c.png)